cpp/include/Ice/Communicator.h

```diff
@@ -6,12 +6,13 @@
 #define ICE_COMMUNICATOR_H
 
 #include "Config.h"
+#include "CommunicatorF.h"
```

Comment:
When we have a F.h file, we should include in the main .h file. This way, no need to redeclare the Ptr.